afk-daily.sh

```diff
@@ -1,1610 +1,1850 @@
 
     # Weekly quests
-    input tap 650 1650 # Weeklies
-    wait
-
-    # Collect weekly Quests
-    getColor 700 670
-    while [ "$RGB" == "82fdf5" ]; do
-        input tap 930 680
-        wait
-        getColor 700 670
-    done
-
-    # Collect weekly Chests
-    input tap 330 430 # Chest 20
-    wait
-    input tap 580 600 # Collect
-    input tap 500 430 # Chest 40
-    wait
-    input tap 580 600 # Collect
-    input tap 660 430 # Chest 60
-    wait
-    input tap 580 600 # Collect
-    input tap 830 430 # Chest 80
-    wait
-    input tap 580 600 # Collect
-    input tap 990 430 # Chest 100
-    wait
-    input tap 580 600 # Collect
-    wait
+    inputTapSleep 650 1650                      # Weeklies
+    collectQuestChests_quick
 
     # Return
-    input tap 70 1650
-    sleep 1
+    inputTapSleep 70 1650 1                     # Return
     verifyRGB 20 1775 d49a61 "Collected daily and weekly quest chests." "Failed to collect daily and weekly quest chests."
 }
 
-# Collects mail
+# ##############################################################################
+# Function Name : collectQuestChests_quick
+# Descripton    : Collects quest chests
+# ##############################################################################
+collectQuestChests_quick() {
+    if [ $DEBUG -ge 4 ]; then echo "[DEBUG] collectQuestChests_quick" >&2; fi
+    # Collect Quests
+    while testColorOR 700 670 82fdf5; do
+        inputTapSleep 930 680
+    done
+
+    if testColorNAND 350 380 54332b; then
+        inputTapSleep 330 430                   # Chest 20
+        inputTapSleep 580 600 0                 # Collect
+    fi
+    if testColorNAND 510 380 543323; then
+        inputTapSleep 500 430                   # Chest 40
+        inputTapSleep 580 600 0                 # Collect
+    fi
+    if testColorNAND 670 380 54331b; then
+        inputTapSleep 660 430                   # Chest 60
+        inputTapSleep 580 600 0                 # Collect
+    fi
+    if testColorNAND 830 380 533323; then
+        inputTapSleep 830 430                   # Chest 80
+        inputTapSleep 580 600 0                 # Collect
+    fi
+    if testColorNAND 1000 380 543323; then
+        inputTapSleep 990 430                   # Chest 100
+        inputTapSleep 580 600                   # Collect
+    fi
+}
+
+# ##############################################################################
+# Function Name : collectMail
+# Descripton    : Collects mail
+# ##############################################################################
 collectMail() {
+    if [ $DEBUG -ge 4 ]; then echo "[DEBUG] collectMail" >&2; fi
     # TODO: I think right here should be done a check for "some resources have exceeded their maximum limit". I have ascreenshot somewhere of this.
-    input tap 960 630
-    wait
-    input tap 790 1470
-    wait
-    input tap 110 1850
-    wait
-    input tap 110 1850
-
-    wait
+    if testColorOR 1020 580 e51f06; then        # Red mark
```

Comment:
This RGB check might also be overkill. It's literally just two button clicks that get spared, it doesn't even take that long. And the number of headaches it can save us might (again) be totally worth it.

I guess stuff like this will be dependant on speed from now on. What would be quicker _and_ more reliable? Checking for an RGB value, even with a `delta`, and acting upon that or performing the task anyways?

Reply:
In my case the script already need more than 20 minutes every day to do everything, so I was looking for a way to avoid useless actions...

Reply:
20 minutes...? What? That's a lot. I don't know how much mine takes, but it's definitely 10 mins maximum. I'd say around 8 minutes. But maybe I'm wrong, I'll take a look at it next time I run it.

Damn, if the script really starts taking this long I guess I will have to take a look into optimizing it eventually...

Reply:
I just ran the script on this PR and it took 31 minutes to finish, which is a lot indeed. Though 60% of that time was probably spent on the King's Tower since today is Sunday and every Factional Tower is open **and** I managed to climb 10 floors on each one... So I'd say the script takes around 18 minutes to finish. Which is still a lot. I'll test this another day out without the King's Tower again and see how long it takes. Maybe I was completely wrong with my perception of how much time this script takes O.o

We should probably implement some type of variable with which we can check if the script should actually fight in the towers or just enter and exit right away, as well as ignore Factional Towers.

Reply:
It's definitely a script to launch and then forget about it while it does its thing. But I didn't imagine it was this bad...

Reply:
With 5 towers, 3 day trik, weekly, wrizz & soren, it tooks me 40min yesterday...

On normal day, 2 towers & wrizz, it's taking 20 min.

<details>
<summary>config.ini</summary>

```ini
# --- CONFIG: Modify accordingly to your game! --- #
# --- Use this link for help: https://github.com/zebscripts/AFK-Daily#configvariables --- #
# Player
canOpenSoren=false
arenaHeroesOpponent=5

# General
waitForUpdate=true
endAt=mail

# Repetitions
maxCampaignFights=10
maxKingsTowerFights=10
totalAmountArenaTries=10
totalAmountTournamentTries=5
totalAmountGuildBossTries=3

# Store
buyStoreDust=true
buyStorePoeCoins=true
buyStorePrimordialEmblem=true
buyStoreAmplifyingEmblem=true
buyStoreSoulstone=true
buyWeeklyGuild=true
buyWeeklyLabyrinth=true

# --- Actions --- #
# Campaign
doLootAfkChest=true
doChallengeBoss=true
doFastRewards=true
doCollectFriendsAndMercenaries=true

# Dark Forest
doSoloBounties=true
doTeamBounties=true
doArenaOfHeroes=true
doLegendsTournament=true
doKingsTower=true

# Ranhorn
doGuildHunts=true
doGuildHuntsBattle=false
doTwistedRealmBoss=true
doBuyFromStore=true
doStrengthenCrystal=true
allowCrystalLevelUp=true
doTempleOfAscension=true
doCompanionPointsSummon=true
# Only works if 'Hide Inn Heroes' is enabled under 'Settings -> Memory'
doCollectOakPresents=true

# End
doCollectQuestChests=true
doCollectMail=true
doCollectMerchantFreebies=true

```

</details>